public/js/logic.js

```diff
@@ -6,35 +6,61 @@ $(document).ready(function(){
 
   // Add a new chest
   $("#add").click(function(){
-    container.append(chest);
-    chestCount++;
+    addChests(1);
   });
 
   // Remove last chest
   $("#remove").click(function(){
-    if (chestCount > 0) {
-      container.children().last().remove();
-      chestCount--;
-    }
+    deleteChests(1);
   });
 
   // Clear all chests
   $("#clear").click(function(){
-    container.empty();
-    chestCount = 0;
+    clearAllChests();
   });
 
-  // Update count when adding, removing, clearing
-  $("#controls button").click(function() {
-    $("#count").val(chestCount);
+  // Remove the chest you click on
+  $("#container").on("click", "div.chest-wrapper", function(){
+    // "on" delegated element events need a static callback function
+    $(this).hide("fast", done());
```

Comment:
We lost the `event` arg in done()

Reply:
Not lost, I removed it. Its deprecated to specify since its implied.

Reply:
Ah. My IDE gave me deprecation warnings about that and my 3am mind didn't figure it out.